client/src/components/Filter.jsx

```diff
@@ -10,8 +24,208 @@ import React from "react";
+        break;
+      default:
+        setCancelButton("Cancel changes");
+    }
+    setButtonStatus(status);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? allFilterData[filterLabel].filterName : undefined;
+  const popOverProps = {
+    sx: {
+      backgroundColor: "gray.50",
+      boxShadow:
+        "0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 2px rgba(0, 0, 0, 0.12), 0px 1px 3px rgba(0, 0, 0, 0.2)",
+      marginTop: "0.5rem",
+      borderRadius: "8px",
+    },
+  };
+  return (
+    <>
+      <Button
+        aria-describedby={id}
+        onClick={handleClick}
+        endIcon={icon}
+        variant="contained"
+        sx={{
+          bgcolor: buttonBgColor,
+          border: "1px solid",
+          borderColor: buttonBorderColor,
+          boxShadow: "none",
+          borderRadius: "8px",
+          textTransform: "none",
+          padding: ".3rem 1rem",
+          color: buttonBorderColor,
+          "&:hover": {
+            bgcolor: buttonBgColor,
+            boxShadow: "none",
+          },
+        }}
+      >
+        <Typography noWrap sx={{ fontSize: "1rem" }}>
+          {allFilterData[filterLabel].filterName}
+        </Typography>
+      </Button>
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        onClose={handleClose}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "left",
+        }}
+        PaperProps={popOverProps}
+      >
+        <FormControl sx={{ mx: 3, my: 1 }}>
+          <FormGroup>
+            {Object.keys(checked).map((key) => (
+              <Grid container key={key} alignItems="center">
+                <Grid item xs>
+                  <FormControlLabel
+                    control={
+                      <Checkbox
+                        checked={checked[key].checked}
+                        onChange={handleChange}
+                        name={key}
+                        sx={{
+                          "&.Mui-checked": {
+                            color: "gray.300",
+                          },
+                        }}
+                      />
+                    }
+                    label={
+                      <Typography noWrap fontSize="1rem" width="160px">
+                        {checked[key].label}
+                      </Typography>
+                    }
+                  />
+                </Grid>
+                <Grid item xs="auto">
+                  <Typography color="gray.300" fontSize="1rem">
+                    {`(${checked[key].count})`}
+                  </Typography>
+                </Grid>
+              </Grid>
+            ))}
+          </FormGroup>
+        </FormControl>
+        <Divider variant="middle" />
+        <Box textAlign="center" py=".5rem">
+          <Button
+            variant="transparent"
+            sx={{ marginRight: ".3rem" }}
+            onClick={handleCancel}
+          >
+            {cancelButton}
+          </Button>
```

Comment:
if I try to use a function the page will not display due to too many re-renders

Reply:
Might not necessarily rule this out though. Next question would be why are there so many rerenders?

This would be dependent on `status` and might be a good lead to asking if that's changing too frequently

Reply:
yeah...I'm not sure why there are so many rerenders...instead i've created an array containing the labels since 0,1,2 from buttonStatus can represent the index of the array